### DISCUSSION TITLE
이미지(업로드 및 unplash) crop 기능

### DISCUSSION THREAD

**Discussion by headring** [Yangjaecheon-Otter-Guardians/simple-thumbnail#114] [Ideas]
### 주제

Features

### Body

이미지를 사용할 시 사용자가 원하는 부분을 crop할 수 있는 기능이 있으면 좋을 것 같아요!
아래는 예시 사진입니다
<img width="417" alt="스크린샷 2023-03-02 오후 10 44 31" src="https://user-images.githubusercontent.com/48895268/222445625-81a1cda5-972e-4184-97ef-3e2caa865a7e.png">

### BDD TASK

랜덤이미지(Unsplash) & 이미지 업로드
- [ ] 이미지 업로드 or 랜덤 이미지 클릭 -> 이미지를 크롭하는 화면이 나탄난다.
- [ ] 화면에서 이미지 크롭을 - or + 형식 또는 드래그(`edited:23.03.04`: crop영역 사용자가 옮기기)하면 -> 미리보기에 어떻게 보일지 나온다
- [ ] 확인을 누르면 -> crop이 된 상태가 적용이되어 나타난다

**Comment by da-in** [comment]
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ 예시 수달 어떡할거에요 진짜 ㅋㅋㅋㅋㅋㅋㅋㅋ

**Reply by caseBread** [reply]
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ 진짜 확 집중돼요

**Comment by caseBread** [comment]
좋은 기능인 것 같습니다.

버튼 하나만 추가하고 모달을 통해 기능을 구현하게되므로 화면이 복잡해질것같지도 않고,
기술적으로도 챌린저블할것같아요!

그리고 BDD관련해서 하나 고민해보고싶은게 있는데,
중앙쪽이 아닌 crop영역을 옮기는 기능을 추가해보는건 어떨까요?
사용자 입장에서 중앙뿐만아니라 사이드쪽의 사진을 크롭해 볼 수도 있겠다 싶어서요!

**Reply by headring** [reply]
crop영역을 옮기는 의견 확인했습니다 :) 
아마 시작한다면 아래 블로그에서 언급된 `react-cropper`를 시도해볼게요!
[블로그](https://velog.io/@bjy100/%EB%A6%AC%EC%95%A1%ED%8A%B8-%EC%83%81%EA%B0%9C%EB%B0%9C%EC%9E%90%ED%8A%B9-%EB%82%B4-%EB%A7%88%EC%9D%8C%EB%8C%80%EB%A1%9C-%EC%9D%B4%EB%AF%B8%EC%A7%80-%EC%9E%98%EB%9D%BC%EC%84%9C-%EC%94%80)
